security/l1-contracts-dev-flow-vision.md

```diff
@@ -0,0 +1,159 @@
+against a tag of the form `op-contracts/vX.Y.Z[-rc.n]`. To deploy the contract, create a `bootstrap`
+command in op-deployer, and run that new command to deploy the new contracts.
+
+An alternate approach that we might support that avoids the need for bootstrap commands is using
+CREATE2 in the `DeploySuperchain.s.sol` and `DeployImplementations.s.sol` scripts. This way,
+deploying contracts is as easy as re-running those scripts—if bytecode for a contract has not changed,
+the resulting create2 address is the same, so we can skip deploying that unchanged contract.
+
+The current [release and versioning process](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/meta/VERSIONING.md)
+is unintuitive and requires a lot of manual work. It is not possible to have the source code for the
+entirety of a release at a single commit, and therefore very difficult to test contract releases.
+Additionally, deploy and upgrade scripts are not part of a release.
+
+Now, OPCM serves as the source of truth for what a release is. A release is defined as the OPCM contract
+itself, along with all contracts that are used in the `deploy()` or `upgrade()` methods of OPCM.
+
+## Execution of Onchain Transactions
+
+With new implementation contracts deployed, we can prepare playbooks, or tasks, in superchain-ops.
+There will be two ways to create tasks: bespoke tasks, and template tasks.
+
+Bespoke tasks are for one-off tasks that are not expected to be repeated. These will be created
+by running a command such as `just new-task {l1ChainId} {taskName}`. This will scaffold a new
+directory. In this directory would be a solidity file that inherits from some base contract. In here
+you write two things: the calls to execute, and validations. The validations are similar to what we
+currently do, and the calls to execute are different—instead of specifying the calldata in an `input.json`
+file, calls are specified in solidity as if you are writing a regular forge script. The tooling
+will extract the sequence of calls from the script and generate the calldata bundle.
+
+Template tasks are for tasks that are expected to be repeated. These will be created by creating
+a solidity template of the same format as a bespoke task, but with some parts replaced with variables.
+These variables are read from a TOML file, and the path to that TOML file is the only input to the
+template script. To create a new task from a template you would run a command such as
+`just new-task-from-template {l1ChainId} {taskName} {templateName}`. This will scaffold a new directory,
+but there will be no solidity file. Instead, a TOML template will be copied into the directory and
+populated as required.
+
+There will also be some other UX and safety improvements for writing these solidity files.
+
+- Addresses for a chain can be fetched in Solidity via `addresses.getAddress("ContractName", l2ChainId)`.
```

Comment:
nit, would it make it better if we used a structs rather than hardcoded strings?

Reply:
I believe this is using the already implemented Addresses.sol from Forge Proposal Simulator. I'll tag the team here to add color.